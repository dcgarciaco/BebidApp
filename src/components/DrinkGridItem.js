import React from 'react';

export const DrinkGridItem = ({ drinkID, drinkName, drinkImageURL }) => {
  return (
    <div className='card animate__animated animate__fadeIn'>
      <img src={drinkImageURL} alt={drinkName} />
      <p>{drinkName}</p>
    </div>
  );
};
