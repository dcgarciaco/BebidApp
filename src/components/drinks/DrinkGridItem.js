import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

export const DrinkGridItem = ({ drinkID, drinkName, drinkImageURL }) => {
  return (
    <Link
      to={`./drink/${drinkID}`}
      className='card animate__animated animate__fadeIn'
    >
      <img src={drinkImageURL} alt={drinkName} />
      <p>{drinkName}</p>
    </Link>
  );
};

DrinkGridItem.propTypes = {
  drinkID: PropTypes.string.isRequired,
  drinkName: PropTypes.string.isRequired,
  drinkImageURL: PropTypes.string.isRequired,
};
