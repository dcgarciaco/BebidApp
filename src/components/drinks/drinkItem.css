.drinkContent {
  display: grid;
  grid-template-columns: 45% 2fr;
  grid-template-rows: 1fr;
  gap: 0px 0px;
  grid-auto-flow: row;
  grid-template-areas: '. .';
  color: #002d29;
}

figure {
  overflow: hidden;
  max-width: 350px;
  justify-self: end;
  align-self: center;
}

.drinkContent img {
  max-width: 100%;
  border-radius: 6px;
}

.drinkInfo {
  margin: 0 auto;
  width: 75%;
}

.drinkContent section {
  align-self: center;
}
