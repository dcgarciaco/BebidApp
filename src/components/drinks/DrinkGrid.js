import React from 'react';
import { useFetchDrinks } from '../../hooks/useFetchDrinks';
import { DrinkGridItem } from './DrinkGridItem';
import PropTypes from 'prop-types';

export const DrinkGrid = ({ category }) => {
  const { data: drinks, loading } = useFetchDrinks(category);

  return (
    <>
      {loading && <p>Loading</p>}

      <section className='card-grid'>
        {drinks.map((drink) => (
          <DrinkGridItem key={drink.drinkID} {...drink} />
        ))}
      </section>
    </>
  );
};

DrinkGrid.propType = {
  category: PropTypes.string.isRequired,
};
