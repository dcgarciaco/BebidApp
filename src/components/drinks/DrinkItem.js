import './drinkItem.css';
import React from 'react';
import { Redirect, useParams } from 'react-router-dom';

import { useFetchDrinkInfo } from '../../hooks/useFetchDrinkInfo';

export const DrinkItem = () => {
  const { drinkID } = useParams();
  

  const { data: drink, loading } = useFetchDrinkInfo(drinkID);
  if (!drink) {
    return <Redirect to='/' />;
  }
  
  

  const renderIngredientList = (data) => {
  
    const listIngredients = data.map((ingredient, index) => (
      <li key={index}>
        {' '}
        {ingredient[1]} {ingredient[0]}{' '}
      </li>
    ));

    return listIngredients;
  };

  if (loading) {
    return <article className=''>{loading && <p>Loading...</p>}</article>;
  } else {
    return (
      <article className='drinkInfo'>
        <div className='drinkContent'>
          <figure>
            <img src={drink[0].drinkImageURL} alt={drink[0].drinkName} />
          </figure>

          <section>
            <h2>{drink[0].drinkName}</h2>
            <ul>{renderIngredientList(drink[0].drinkIngredients)}</ul>
            <p>
              <b>Glass</b> {drink[0].drinkGlass}
            </p>
            <h3>Instructions</h3>
            <p>{drink[0].drinkInstructions}</p>
          </section>
        </div>
      </article>
    );
  }
};
