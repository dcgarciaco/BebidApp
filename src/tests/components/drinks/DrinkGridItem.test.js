import { shallow } from 'enzyme';
import { DrinkGridItem } from '../../../components/drinks/DrinkGridItem';

describe('Test in <DrinkGridItem />', () => {
  const drinkID = '10';
  const drinkImageURL = 'https://localhost/drink.jpg';
  const drinkName = 'Titulo de imagen';
  const wrapper = shallow(
    <DrinkGridItem
      drinkID={drinkID}
      drinkName={drinkName}
      drinkImageURL={drinkImageURL}
    />
  );

  test('Should show the component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Should have a paragraph with the image tile', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(drinkName);
  });

  test('should have an image with the same url and alt of props', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(drinkImageURL);
    expect(img.prop('alt')).toBe(drinkName);
  });

  test('should have animate_fadeIn class', () => {
    const Link = wrapper.find('Link');
    const className = Link.prop('className');

    expect(className.includes('animate__fadeIn')).toBe(true);
  });

  test('should have a link to the drink with the ID', () => {
    const link = wrapper.find('Link');
    expect(link.prop('to')).toBe('./drink/' + drinkID);
  });
});
