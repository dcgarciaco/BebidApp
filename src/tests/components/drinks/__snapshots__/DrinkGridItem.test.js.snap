// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <DrinkGridItem /> Should show the component correctly 1`] = `
<Link
  className="card animate__animated animate__fadeIn"
  to="./drink/10"
>
  <img
    alt="Titulo de imagen"
    src="https://localhost/drink.jpg"
  />
  <p>
    Titulo de imagen
  </p>
</Link>
`;
